client/src/get-components-data.js

```diff
@@ -5,11 +5,103 @@ var request = require('minimal-request');
 
 var settings = require('./settings');
 var _ = require('./utils/helpers');
+var url = require('url');
 
 module.exports = function(config){
 
-  return function(toDo, options, cb){
+  var handleErrorResponse = function(requestDetails, error, response, components) {
+    var errorDetails = error ? error.toString() : undefined;
+
+    if (response && response.error) {
+      if (errorDetails) {
+        errorDetails = format('{0} {1}', errorDetails, response.error); 
+      } else {
+        errorDetails = response.error;
+      }
+    }
+    
+    if (!errorDetails) {
+      errorDetails = settings.emptyResponse;
+    }
+
+    var responses = [];
+    _.each(components, function(){
+      responses.push({
+        response: {
+          error: format(settings.connectionError, JSON.stringify(requestDetails), errorDetails)
+        }
+      });
+    });
+    
+    return responses;
+  };
+
+  var performPost = function(endpoint, serverRendering, options, callback) {
+    var requestDetails = {
+      url: endpoint,
+      method: 'post',
+      headers: options.headers,
+      timeout: options.timeout,
+      json: true,
+      body: {
+        components: serverRendering.components,
+        parameters: options.parameters || {}
+      }
+    };
+
+    request(requestDetails, function(error, responses) {
+      if(!!error || !responses || _.isEmpty(responses)){
+        responses = handleErrorResponse(requestDetails, error, responses, serverRendering.components);
+      }
+
+      callback(responses);
+    });
+  };
+
+  var performGet = function(endpoint, serverRendering, options, callback) {
+    var component = serverRendering.components[0];
+    var urlPath = component.name + (component.version ? '/' + component.version : '');
+    var queryString;
+
+    if (options.parameters) {
```

Comment:
We have an `href-builder` for building the client-side failover urls. Can we move this there perhaps in a new "server" method?
https://github.com/opentable/oc/blob/master/client/src/href-builder.js#L9

I think you may be 1) able to reuse most of that stuff and perhaps refactor a bit as I think the only thing changing is the base url 2) keep the same convention which is version/?params instead of version??params